src/sentry/incidents/models/incident.py

```diff
@@ -97,28 +108,33 @@ def get_active_incident(self, alert_rule, project):
 
     @classmethod
     def clear_active_incident_cache(cls, instance, **kwargs):
+        # instance is an Incident
         for project in instance.projects.all():
-            cache.delete(cls._build_active_incident_cache_key(instance.alert_rule_id, project.id))
-            assert (
-                cache.get(cls._build_active_incident_cache_key(instance.alert_rule_id, project.id))
-                is None
-            )
+            subscription = instance.subscription
+            if subscription:
+                key = cls._build_active_incident_cache_key(
+                    instance.alert_rule_id, project.id, subscription.id
+                )
+            else:
+                key = cls._build_active_incident_cache_key(instance.alert_rule_id, project.id, None)
+            cache.delete(key)
+            assert cache.get(key) is None
 
     @classmethod
     def clear_active_incident_project_cache(cls, instance, **kwargs):
-        cache.delete(
-            cls._build_active_incident_cache_key(
-                instance.incident.alert_rule_id, instance.project_id
+        # instance is an IncidentProject
+        project_id = instance.project_id
+        if instance.incident.subscription:
+            subscription_id = instance.incident.subscription_id
+            key = cls._build_active_incident_cache_key(
+                instance.incident.alert_rule_id, project_id, subscription_id
             )
-        )
-        assert (
-            cache.get(
-                cls._build_active_incident_cache_key(
-                    instance.incident.alert_rule_id, instance.project_id
-                )
+        else:
+            key = cls._build_active_incident_cache_key(
+                instance.incident.alert_rule_id, project_id, None
             )
-            is None
-        )
+        cache.delete(key)
+        assert cache.get(key) is None
```

Comment:
I would be careful with this as it could fail from concurrent requests that re-create the cache key.

Reply:
hmmm yea
this was previously here though 🤔 

maybe i'll leave it for now?